src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt

```diff
@@ -374,23 +375,18 @@ internal class RecommendationService(
           sensitive = considerationSensitive,
         ),
       )
-
-      sendConsiderationRationaleDomainEvent(recommendationId, existingRecommendationEntity, userId)
     }
 
-    val sendSpoDeleteRationaleToDelius =
-      requestJson.has("sendSpoDeleteRationaleToDelius") && requestJson.getBoolean("sendSpoDeleteRationaleToDelius")
-
+    val sendSpoDeleteRationaleToDelius = requestJson.has("sendSpoDeleteRationaleToDelius") &&
+      requestJson.get("sendSpoDeleteRationaleToDelius").asBoolean()
     if (sendSpoDeleteRationaleToDelius) {
-      log.info("send spo delete recommendation rationale to delius")
       existingRecommendationEntity.data = existingRecommendationEntity.data.copy(
         spoDeleteRecommendationRationale = updatedRecommendation.spoDeleteRecommendationRationale,
         sendSpoDeleteRationaleToDelius = updatedRecommendation.sendSpoDeleteRationaleToDelius,
       )
-      sendDeleteRecommendationDomainEvent(recommendationId, existingRecommendationEntity, userId)
     }
 
-    existingRecommendationEntity.deleted = requestJson.has("status") && requestJson.getString("status") == "DELETED"
+    existingRecommendationEntity.deleted = requestJson.has("status") && requestJson.get("status").asText() == "DELETED"
```

Comment:
Cool trick here when modifying multiple fields of an object, you can use the `apply` [scope function](https://kotlinlang.org/docs/scope-functions.html) to avoid repetition

```kotlin
    existingRecommendationEntity.apply {
      deleted = requestJson.has("status") && requestJson.get("status").asText() == "DELETED"
      data.recallConsideredList = updateRecallConsideredList(
        updatedRecommendation,
        data,
        userId,
        readableUserName,
      )
      data = updatePageReviewedValues(updatedRecommendation, existingRecommendationEntity).data
      refreshData(pageRefreshIds, data)
    }
```

Reply:
Nice, I like.